libzim/libzim.pyx

```diff
@@ -269,72 +281,110 @@ cdef class _Creator:
+        """Set size of created clusters.
+
+        Check libzim for default setting. (Fall 2021 default: 2Mib).
+
+        libzim will store at most this value per cluster before creating
+        another one.
 
-            libzim will store at most this value per cluster before creating
-            another one.
-            Fall 2021 default: 2Mib"""
+        Args:
+            size (int): The maximum size (in bytes) for each cluster.
+
+        Returns:
+            The Creator instance with updated cluster size settings.
+        """
         if self._started:
             raise RuntimeError("Creator started")
         self.c_creator.configClusterSize(size)
         return self
 
     def config_indexing(self, bool indexing: bool, str language: str) -> Creator:
-        """Configure fulltext indexing feature
+        """Configures the full-text indexing feature.
+
+        Args:
+            indexing (bool): whether to create a full-text index of the content
+            language (str): language (ISO-639-3 code) to assume content in during indexation.
 
-            indexing: whether to create a full-text index of the content
-            language: language (ISO-639-3 code) to assume content in during indexation"""
+        Returns:
+            The Creator instance with updated indexing settings.
+        """
         if self._started:
             raise RuntimeError("Creator started")
         self.c_creator.configIndexing(indexing, language.encode('UTF-8'))
         return self
 
     def config_nbworkers(self, int nbWorkers: pyint) -> Creator:
-        """Number of thread to use for internal worker"""
+        """Configures the number of threads to use for internal workers.
+
+        Args:
+            nbWorkers (int): The number of threads to allocate.
+
+        Returns:
+            The Creator instance with updated worker thread settings.
+        """
         if self._started:
             raise RuntimeError("Creator started")
         self.c_creator.configNbWorkers(nbWorkers)
         return self
 
     def set_mainpath(self, str mainPath: str) -> Creator:
-        """Set path of the main entry"""
+        """Set path of the main entry.
+
+        Args:
+            mainPath (str): The path of the main entry.
+
+        Returns:
+            The Creator instance with the updated main entry path.
+        """
         self.c_creator.setMainPath(mainPath.encode('UTF-8'))
         return self
 
     def add_illustration(self, int size: pyint, content: bytes):
-        """Add a PNG illustration to Archive
+        """Add a PNG illustration to Archive.
 
-            https://wiki.openzim.org/wiki/Metadata
+        Refer to https://wiki.openzim.org/wiki/Metadata for more details.
 
-            Raises
-            ------
-                RuntimeError
-                    If an Illustration exists with the same size"""
+        Args:
+            size (int): The size of the illustration in bytes.
```

Comment:
Size is actually the width of the PNG image being added. This is assumed to be a square image sor a 48x48px image would use `size=48` here.